sdk/servicebus/Azure.Messaging.ServiceBus/src/Addons/ServiceBusClientExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Azure.Messaging.ServiceBus.Addons
+{
+    /// <summary>
+    /// These extensions allow additional functionality for the <see cref="ServiceBusClient"/>.
+    /// </summary>
+    public static class ServiceBusClientExtensions
+    {
+        /// <summary>
+        /// Creates a <see cref="ServiceBusMessage"/> with the AMQP Data body type.
+        /// </summary>
+        /// <param name="data">The data to </param>
+        /// <returns></returns>
+        public static ServiceBusMessage CreateAmqpDataMessage(ReadOnlyMemory<byte> data) =>
```

Comment:
Since this is meant for scenarios where users want to take full advantage of the AMQP spec for the body, I wonder if we should mirror the spec exactly and have this take an `IEnumerable<ReadOnlyMemory<byte>>`

Reply:
Not sure that you're referring to the spec rather than the particular implementation of the AMQP library in this case.    As far as I recall, the spec just calls for a blob of bytes - the `Microsoft.Azure.Amqp` library decided to use an enumerable of read-only segments. 

There's nothing that `IEnumerable<ReadOnlyMemory<byte>>` can express that `ReadOnlyMemory<byte>` can't - it would just make it a bit easier to pass a series of segments from different sources.  I don't think that is a scenario common enough to offset the annoyance of having to allocate an enumerable for the more common case.

Reply:
The spec says this:
"The body consists of one of the following three choices: one or more data sections, one or more amqp-sequence sections, or a single amqp-value section."

Since this is for advanced scenarios, we may as well match it exactly.

Reply:
I stand corrected; I was thinking of this as the value and not the data section.   Fair enough.  We'll have to make sure that when the `AmqpMessageConverter` does its work, it knows to emit multiple sections.  Right now it would cram them all into one.

Reply:
I will work on this in the morning. This takes a bit more effort since there would no longer be a 1-1 conversion between `BinaryData` and the `Data` body.

Reply:
This took a bit. Should be implemented now.